doc/arch/adr-007-stackdriver.md

```diff
@@ -0,0 +1,62 @@
+
+Regardless of if the systems are hosted on premise or as a service some costs of ownership still remain. One significant cost among these is the cost of knowing and utilizing the tools that need to be distributed among team members. It is for us imperative to keep the number of services and tools (especially ones that are in some way overlapping) to a minimum.
+
+In the case of monitoring the recommendation from ops engineering and others has been to use [Google Stackdriver](https://cloud.google.com/stackdriver/). The reasons being that it's a very complete monitoring, logging, error reporting and alerting solution natively integrated with Google Cloud Platform, as well as it's quite simple and cheap compared to all alternatives.
+
+Stackdriver is already available in our Kubernetes cluster for the other services in Akvo Lumen. Choosing a different monitoring solution for Windshaft would add a learning cost and also introduce risks that the individual monitoring systems would need to share the attention of the responsible system owner, thus increasing the risk of fatal problems going unnoticed.
+
+### Considerations
+Using a third party service provider for machine and application monitoring still comes with some considerations that need to be seen and addressed. These are:
+
+* Data security
+* Avoiding vendor lock-in
+* Cost
+
+### Data security
+First and foremost, it should be possible to switch monitoring on and off. Both for the sake of running local development environments without monitoring having to be set up, but also in the cases where partners or other users want to run Lumen by themselves and not having to pay for Google Stackdriver usage. Being able to switch off application monitoring is also a first, minimum step to enable Lumen to be run within the confines of the borders of a country.
+
+It is also important that care is taken to ensure that data that is sent from Windshaft to Stackdriver does not identify the partner or leak any of the data on their subjects.
+
+To allow for efficient usage and to get value from the monitoring system, an application owner needs to be able to identify different cases of use to see which cases have a severe impact on application performance. A suggested strategy for achieving this is to use a strong hashing algorithm on the identifiers in the monitoring data (for example the tenant and user IDs) and to generate and store the hash keys, the salt(s) and the identity lookup table privately and securely for our convenience. Even when we obfuscate the identities in this way, we should still carefully consider what data and metrics will be sent to the monitoring systems so that these could not be used to glean important information on our users and their source data.
+
+Question: how will SQL queries be handled in the monitoring system? [Dan]
```

Comment:
My concern was around what kind of information will be visible in the monitoring system with regards to the SQL queries. If these queries are not visible (with query parameters) then I'm ok with it.

I agree with you that the exact time elapsed is of less importance as long as we have some numbers that are consistently created and comparable to each other.